x-pack/plugins/reporting/server/browsers/chromium/driver_factory/metrics.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Metrics as PuppeteerMetrics } from 'puppeteer';
+import { cpus } from 'os';
+
+declare module 'puppeteer' {
+  interface CDPSession {
+    send(command: 'Performance.getMetrics'): Promise<RawMetrics>;
+  }
+}
+
+type RawMetrics = Metrics;
+
+export interface Metrics {
+  metrics: Metric[];
+}
+
+interface Metric {
+  name: keyof NormalizedMetrics;
+  value: unknown;
+}
+
+interface NormalizedMetrics extends PuppeteerMetrics {
+  ProcessTime: number;
+}
+
+interface PerformanceMetrics {
+  cpu: number;
+  cpuInPercentage: number;
+  memory: number;
+  memoryInMegabytes: number;
+}
+
+function normalizeMetrics({ metrics }: Metrics) {
+  return (Object.fromEntries(
+    metrics.map(({ name, value }) => [name, value])
+  ) as unknown) as NormalizedMetrics;
+}
+
+function getCpuUsage(start: NormalizedMetrics, end: NormalizedMetrics) {
+  return (end.ProcessTime - start.ProcessTime) / (end.Timestamp - start.Timestamp) / cpus().length;
```

Comment:
What is the difference between `x.ProcessTime` and `x.Timestamp`?

Reply:
`ProcessTime` is the amount of time for which a CPU was used in the current process ([Wikipedia](https://en.wikipedia.org/wiki/CPU_time)). It is a sum of the time from all the CPU cores that have been used.
`Timestamp` is just the current time.

Sometimes when the process was in the idle state, the `ProcessTime` will be less than the actual time the process was running for. Sometimes, when the process uses multiple cores, the `ProcessTime` will be more than the actual duration.